dlp/snippets/src/main/java/dlp/snippets/ReidentifyWithDeterministicEncryption.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dlp.snippets;
+
+// [START dlp_reidentify_deterministic]
+
+import com.google.cloud.dlp.v2.DlpServiceClient;
+import com.google.privacy.dlp.v2.ContentItem;
+import com.google.privacy.dlp.v2.CryptoDeterministicConfig;
+import com.google.privacy.dlp.v2.CryptoKey;
+import com.google.privacy.dlp.v2.CustomInfoType;
+import com.google.privacy.dlp.v2.DeidentifyConfig;
+import com.google.privacy.dlp.v2.InfoType;
+import com.google.privacy.dlp.v2.InfoTypeTransformations;
+import com.google.privacy.dlp.v2.InspectConfig;
+import com.google.privacy.dlp.v2.KmsWrappedCryptoKey;
+import com.google.privacy.dlp.v2.LocationName;
+import com.google.privacy.dlp.v2.PrimitiveTransformation;
+import com.google.privacy.dlp.v2.ReidentifyContentRequest;
+import com.google.privacy.dlp.v2.ReidentifyContentResponse;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.apache.commons.codec.binary.Base64;
+
+public class ReidentifyWithDeterministicEncryption {
+
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+
+    // The Google Cloud project id to use as a parent resource.
+    String projectId = "your-project-id";
+    // The string to re-identify
+    String textToReIdentify = "My SSN is SSN_TOKEN(36):Abmg1/Wql5D6stA1s467OQsGyegFrV0Qe/E=";
```

Comment:
this encrypted text only works with the wrappedkey you used to generate it... it will fail if someone uses a different wrapped key... can we add a DeidentifyWithDeterministicEncryption class, and use that here to generate this text?

Reply:
I see the PR for deid with deterministic encryption here: https://github.com/GoogleCloudPlatform/java-docs-samples/pull/7848

Reply:
Thank you for reviewing the sample. According to me, this is a re-identification code sample, and for the creation of encrypted text, the user can refer to a de-identification code sample. Additionally, we have not used the de-identification class for the other reidentification code provided [here](https://cloud.google.com/dlp/docs/samples). With that said, please let me know how should I proceed with this and any other re-identification examples.

Reply:
I see - python embeds the token here: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/636039e54c227629988838155ee8c9345e66bd13/dlp/snippets/deid_test.py#L236
It also embeds the wrapped and unwrapped, which won't work if someone uses a different google project (e.g. an external contributor).

I also see you have used the deid code to generate the input for reid. I think that is safer and more portable.